module-apps/apps-common/windows/OptionWindow.hpp

```diff
@@ -31,5 +31,11 @@ namespace gui
         void onClose(CloseReason reason) override;
         void rebuild() override;
         void buildInterface() override;
+
+      protected:
+        /// this have to be called in inheriting window if someone passes callback to it
+        /// othervise destruction order will cause call to non existing parent element
+        /// this is because on how OptionWindows are designed
+        void destroyForTheFuture();
```

Comment:
almost like back to the future :D

Reply:
feel free to provide me a better name - it will be appreciated!  (there are better ways to fix this issue, none is that short and non-intrusive though)